chrysom/listenerClient.go

```diff
@@ -38,88 +39,108 @@ const (
 )
 
 // ListenerConfig contains config data for polling the Argus client.
-type ListenerClientConfig struct {
-	// Listener provides a mechanism to fetch a copy of all items within a bucket on
-	// an interval.
+type ListenerClientIn struct {
+	fx.In
+
+	// Listener fetches a copy of all items within a bucket on
+	// an interval based on `BasicClientConfig.PullInterval`.
 	// (Optional). If not provided, listening won't be enabled for this client.
 	Listener Listener
-
-	// PullInterval is how often listeners should get updates.
-	// (Optional). Defaults to 5 seconds.
-	PullInterval time.Duration
-
-	// Logger to be used by the client.
-	// (Optional). By default a no op logger will be used.
-	Logger *zap.Logger
+	// Config configures the ancla client and its listeners.
+	Config BasicClientConfig
+	// PollsTotalCounter measures the number of polls (and their success/failure outcomes) to fetch new items.
+	PollsTotalCounter *prometheus.CounterVec `name:"chrysom_polls_total"`
+	// Reader is the DB interface used to fetch new items using `GeItems`.
+	Reader Reader
+	// GetLogger returns a logger from the given context.
+	GetLogger GetLogger
+	// SetLogger embeds the `Listener.logger` in outgoing request contexts for `Listener.Update` calls.
+	SetLogger SetLogger
 }
 
 // ListenerClient is the client used to poll Argus for updates.
 type ListenerClient struct {
 	observer  *observerConfig
-	logger    *zap.Logger
```

Comment:
listener client will get a logger from a context instead (keeping it consistent with BasicClient)